k4EDM4hep2LcioConv/src/k4Lcio2EDM4hepConv.cpp

```diff
@@ -605,14 +605,21 @@ namespace LCIO2EDM4hepConv {
     return headerColl;
   }
 
-  podio::Frame convertEvent(EVENT::LCEvent* evt)
+  podio::Frame convertEvent(EVENT::LCEvent* evt, const std::vector<std::string>& collsToConvert)
   {
     auto typeMapping = LcioEdmTypeMapping {};
     std::vector<CollNamePair> edmevent;
     std::vector<std::pair<std::string, EVENT::LCCollection*>> LCRelations;
-    const auto& lcnames = evt->getCollectionNames();
+
+    const auto& lcnames = [&collsToConvert, &evt]() {
+      if (collsToConvert.empty()) {
+        return *evt->getCollectionNames();
+      }
+      return collsToConvert;
+    }();
+
     // In this loop the data gets converted.
-    for (const auto& lcioname : *lcnames) {
+    for (const auto& lcioname : lcnames) {
       const auto& lcioColl = evt->getCollection(lcioname);
```

Comment:
We don't, but I should document the assumption that `collsToConvert` should only contain collections that are present in the event. For the standalone `lcio2edm4hep` we have that guarantee, because we either only convert complete events, or we patch missing collections in before we call `convertEvent`.